### PR TITLE
Add keyword fields above `ignore_above` to `_ignored`

### LINKED ISSUE #74228: Add keyword fields above `ignore_above` to the `_ignored` field
Currently the `_ignore` field indexes and stores the names of every field in a document that has been ignored
because it was malformed and ignore_malformed was turned on. The goal is to give the user a means to identify documents
where values have been dropped and warn that these values are not searchable or aggregatable, althought they might be
present in the "_source".
Since the `ignore_above` option for keyword fields serves a somewhat similar purpose (drop values if they exceed a certain length, mostly as a safeguard for the system), we should also add these fields to the "_ignored" field.

Relates to #74121

### PR DESCRIPTION
**Pull request by cbuescher**
Currently the `_ignore` field indexes and stores the names of every field in a document that has been ignored
because eg. it was malformed. The `ignore_above` option for keyword-type fields
serves a somewhat similar purpose, so this change add logix that  adds these
fields to the "_ignored" field as well for `keyword`, `wildcard` and
`icu_collation_keyword` fields.

Closes #74228

### CODE REVIEW

**Comment by cbuescher** [comment]
When looking through the field types that support "ignore_above" I also found that "flattened" has an option for this, however the logic which we use there is different from other keyword-type fields and I'd like to look at whether we want similar behaviour there more closely in a separate issue which I'm going to open for tracking.

File: x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java [hunk 1]
```diff
@@ -964,14 +964,17 @@ protected void parseCreateField(ParseContext context) throws IOException {
         ParseContext.Document parseDoc = context.doc();
 
         List<IndexableField> fields = new ArrayList<>();
-        createFields(value, parseDoc, fields);
+        if (value != null && value.length() <= ignoreAbove) {
```
Comment (ywelsch) [inline x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java:967]: perhaps pull out the null check, i.e.,

```
if (value != null) {
  if (value.length() > ignoreAbove) {
    context.addIgnoredField(name());
  } else {
    createFields(value, parseDoc, fields);
  }
}
```

**Review by ywelsch — APPROVED** [review]
I've left one nit, o.w. LGTM.

**Comment by cbuescher** [comment]
@ywelsch thanks for reviewing

**Pull request merged**